hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/scale/AbstractSTestS3AHugeFiles.java

```diff
@@ -257,11 +309,28 @@ public void test_010_CreateHugeFile() throws IOException {
     verifyStatisticGaugeValue(iostats, putRequestsActive.getSymbol(), 0);
     verifyStatisticGaugeValue(iostats,
         STREAM_WRITE_BLOCK_UPLOADS_BYTES_PENDING.getSymbol(), 0);
+
     progress.verifyNoFailures(
         "Put file " + fileToCreate + " of size " + filesize);
-    if (streamStatistics != null) {
-      assertEquals("actively allocated blocks in " + streamStatistics,
-          0, streamStatistics.getBlocksActivelyAllocated());
+    assertEquals("actively allocated blocks in " + streamStatistics,
+        0, streamStatistics.getBlocksActivelyAllocated());
+
+    if (!expectImmediateFileVisibility()) {
```

Comment:
Just an idea. Why don't we put this whole block of code as part of ITestS3AHugeMagicCommits?

Reply:
Once the file has been created and all the basic validations done..add for extra validations depending on the subclassed tests.

Reply:
I will add an override point; will need to pass in stream stats, path and bytes read +anything else

Reply:
actually, can do it all in the test_030 asserts if I make the filesize field visible through a getter.